### ISSUE TITLE
Sidecar Injection with annotation fails with error.

### ISSUE THREAD

**Issue by shamsher31** [istio/istio.io#8077]
Ref: https://preliminary.istio.io/latest/docs/setup/additional-setup/sidecar-injection/#policy

Fails with following error.
```
error: error validating "STDIN": error validating data: ValidationError(Deployment.spec):
missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec;
if you choose to ignore these errors, turn validation off with --validate=false
```

Using this YAML.

```yaml
> apiVersion: apps/v1
> kind: Deployment
> metadata:
>   name: ignored
> spec:
>   template:
>     metadata:
>       annotations:
>         sidecar.istio.io/inject: "false"
>     spec:
>       containers:
>       - name: ignored
>         image: governmentpaas/curl-ssl
>         command: ["/bin/sleep","infinity"]
```

Where the `selector` and `labels` are missing.

**Comment by ericvn** [comment]
Not sure if a fix for https://github.com/istio/istio.io/issues/8057 could also be added here.

**Comment by shamsher31** [comment]
@ericvn The uninstalling sidecar injection throws the following error.

```
$ kubectl delete mutatingwebhookconfiguration istio-sidecar-injector
mutatingwebhookconfiguration.admissionregistration.k8s.io "istio-sidecar-injector" deleted

[shaansar@master-node istio.io]$ kubectl -n istio-system delete service istio-sidecar-injector
Error from server (NotFound): services "istio-sidecar-injector" not found

[shaansar@master-node istio.io]$ kubectl -n istio-system delete deployment istio-sidecar-injector
Error from server (NotFound): deployments.apps "istio-sidecar-injector" not found

[shaansar@master-node istio.io]$ kubectl -n istio-system delete serviceaccount istio-sidecar-injector-service-account
Error from server (NotFound): serviceaccounts "istio-sidecar-injector-service-account" not found

[shaansar@master-node istio.io]$ kubectl delete clusterrole istio-sidecar-injector-istio-system
Error from server (NotFound): clusterroles.rbac.authorization.k8s.io "istio-sidecar-injector-istio-system" not found

[shaansar@master-node istio.io]$ kubectl delete clusterrolebinding istio-sidecar-injector-admin-role-binding-istio-system
Error from server (NotFound): clusterrolebindings.rbac.authorization.k8s.io "istio-sidecar-injector-admin-role-binding-istio-system" not found
```

I'm thinking if [`kube-uninject`](https://github.com/istio/istio/blob/707b28d8b8e6dac582f55fe40403109f4d3cbb4e/istioctl/cmd/kubeuninject.go#L344) will be useful here in any way?

**Comment by shamsher31** [comment]
> Not sure if a fix for #8057 could also be added here.

I have removed unused delete commands.
